### DISCUSSION TITLE
recording rtsp streams and rtmp streams

### DISCUSSION THREAD

**Discussion by MissingName01** [bluenviron/mediamtx#1787] [Questions] [answered]
im running my mediamtx server right now and under the runOnReady parameter im recording all incoming streams 
using

 ffmpeg -i rtsp://localhost:$RTSP_PORT/$RTSP_PATH -c copy -f segment -strftime 1 -segment_time 900 -segment_format mpegts "%Y-%m-%d_%H-%M-%S_%s.ts"

is there a way to have the RTSP_PATH included in the saved file since i have multiple streams coming in that i need to record and the cameras are constantly changing IPs or paths
also is it possible to dictate what directory the .ts files will be saved?

**Comment by dbuscaino** [comment]
Hi, for the path you just need to specify the absolute path of the directory in witch you want to save the videos. For example in linux:
`ffmpeg -i rtsp://localhost:$RTSP_PORT/$RTSP_PATH -c copy -f segment -strftime 1 -segment_time 900 -segment_format mpegts /home/ubuntu/my_video/%Y-%m-%d_%H-%M-%S_%s.ts`

To add the rtsp path's name to the file I think you can try something like that:
`ffmpeg -i rtsp://localhost:$RTSP_PORT/$RTSP_PATH -c copy -f segment -strftime 1 -segment_time 900 -segment_format mpegts /home/ubuntu/my_video/$RTSP_PATH-%Y-%m-%d_%H-%M-%S_%s.ts`

**Reply by MissingName01** [reply]
So the file path part works but if I type i use 

/Videos/$RTSP_PATH-%Y-%m-%d_%H-%M-%S_%s.ts

The file won't save anywhere, I also tried adding {} around the RTSP_PATH since I saw that somewhere but still no luck

**Reply by dbuscaino** [reply, marked as answer]
Well it's strange... I tried my command on linux with absolute path and it works. Which OS are you using? Is `/Videos/$RTSP_PATH-%Y-%m-%d_%H-%M-%S_%s.ts` an absolute path?
Try `/Videos/%Y-%m-%d_%H-%M-%S_%s-$RTSP_PATH.ts` or `/Videos/path:$RTSP_PATH-%Y-%m-%d_%H-%M-%S_%s.ts` maybe it's just a placeholder position problem but i don't think so

**Reply by MissingName01** [reply]
I'm running this on ubuntu but will endure switching to either rocky or centos7 most likely.

Right now though with having the $RTSP_PATH 
when I show status it sees the rtsp path but says 
"Could not write header for output file #0 (incorrect codec parameter ?): no such file or directory
It gives this after trying to open the file path with the rtsp path at the end

**Reply by MissingName01** [reply]
OK so I found the issue, ffmpeg is unable to create new directories, and my rtsp path has "live/alias" in it, my solution to it was to create a file called "live" within the Videos directory and this solved it

[a comment by an automated account is not shown]
